src/autopas/containers/octree/OctreeDirection.h

```diff
@@ -0,0 +1,532 @@
+ *
+ * @return A pointer to a static table. The last element is the "invalid" edge OO.
+ */
+inline Edge *getEdges() {
+  static Edge table[] = {LD, LU, LB, LF, RD, RU, RB, RF, DB, DF, UB, UF, OO};
+  return table;
+}
+
+/**
+ * Get all available vertices for a cube.
+ *
+ * @return A pointer to a static table. The last element is the "invalid" vertex OOO.
+ */
+inline Vertex *VERTICES() {
+  static Vertex table[] = {LDB, LDF, LUB, LUF, RDB, RDF, RUB, RUF, OOO};
+  return table;
+}
+
+/**
+ * Map an arbitrary vertex to a flat index.
+ *
+ * @param vertex An element from the Vertex enum
+ * @return A flat index in the range of 0 to 7 for any valid vertex. For invalid input -1 is returned.
+ */
+inline int vertexToIndex(Vertex vertex) {
+  // @todo This is very slow and could be sped up.
+  //   Mentioned in https://github.com/AutoPas/AutoPas/issues/623
+  int result = -1;
+  for (int i = 0; i < 8; ++i) {
+    if (vertex == VERTICES()[i]) {
+      result = i;
+      break;
+    }
+  }
+  return result;
+}
+
+/**
+ * A vertex is also capable of specifying an arbitrary octant in 3D.
+ */
+using Octant = Vertex;
+
+/**
+ * Check if a given list of elements contains the element to test.
+ *
+ * @tparam T The base type of the list elements
+ * @param all A pointer to the start element in the list
+ * @param stop A pointer to the element which is one past the last element in the list
+ * @param test The element to check
+ * @return true if the list contains the test element, false otherwise
+ */
+template <typename T>
+inline bool contains(T *all, T stop, Any test) {
+  bool result = false;
+  for (T *t = all; *t != stop; ++t) {
+    if (test == *t) {
+      result = true;
+      break;
+    }
+  }
+  return result;
+}
+
+/**
+ * Check if f is a face.
+ *
+ * @tparam T
+ * @param f The parameter to check
+ * @return true iff f is in the list returned from getFaces()
+ */
+template <typename T>
+inline bool isFace(T f) {
+  return contains(getFaces(), O, f);
+}
+
+/**
+ * Check if e is an edge.
+ *
+ * @tparam T
+ * @param e The parameter to check
+ * @return true iff e is in the list returned from getEdges()
+ */
+template <typename T>
+inline bool isEdge(T e) {
+  return contains(getEdges(), OO, e);
+}
+
+/**
+ * Check if v is a vertex.
+ *
+ * @tparam T
+ * @param v The parameter to check
+ * @return true iff v is in the list returned from VERTICES()
+ */
+template <typename T>
+inline bool isVertex(T v) {
+  return contains(VERTICES(), OOO, v);
+}
+
+/**
+ * This function implements a LUT obtained from the Samet paper:
+ * "ADJ(I,O) is true if and only if octant O is adjacent to the Ith face, edge, or vertex of O's containing block"
+ *
+ * @param direction The direction I to search the adjacent neighbor
+ * @param octant The octant O from the paper
+ * @return true if the neighbor is adjacent, false otherwise
+ */
+inline bool ADJ(Any direction, Vertex octant) {
+  static std::array<std::array<bool, 8>, 1 << 9> table;
```

Comment:
this table too could be a static member in the namespace. Actually you could also create an nameless namespace within the namespace. Everything in there is "private" for the stuff in this file.